aten/src/ATen/cpu/vec256/vec256_qint.h

```diff
@@ -1093,10 +1093,19 @@ struct Vec256QuantizedConverter {
       Vec256<float> scale_zp_premul) const {
     float_vec_return_type rv;
     for (int i = 0; i < float_num_vecs(); ++i) {
+      float tmp_vals[8];
       for (int j = 0; j < 8; ++j) {
-        rv[i][j] = at::native::dequantize_val<T>(
+        tmp_vals[j] = at::native::dequantize_val<T>(
             scale[j], zero_point[j], T(vals[8 * i + j]));
       }
+      rv[i] = Vec256<float>(tmp_vals[0],
+          tmp_vals[1],
+          tmp_vals[2],
+          tmp_vals[3],
+          tmp_vals[4],
+          tmp_vals[5],
+          tmp_vals[6],
+          tmp_vals[7]);
```

Comment:
Why did this change?

Reply:
So what happens is that the use of `Vec256QuantizedConverter` is only for non-AVX path, which is the case for us. This change is for dequantize util. Now Vec256<float> class that is not specialized for AVX provides `[]` operator here: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cpu/vec256/vec256_base.h#L131, so that you can effectively update the values held by Vec256<float> class directly. So the values held by vec256<float> type are accessed as a pointer and can be dereferenced. However Vec256<float> implementation for ARM, similar to AVX cannot provide a way to directly modify particular element of the vector datatype it holds by providing a pointer that can be dereferenced. That is why this change. And this leads to the change above.